### PR TITLE
Add option to click on a row to open editor

### LINKED ISSUE #168: Add option to open editor by row click

### PR DESCRIPTION
**Pull request by DiegoCardoso**
With `<vaadin-crud click-row-to-edit>`, clicking on a grid row opens the editor.

- A second click/tap on the same item closes editor (for `editor-position="bottom|aside"` only)
- When enabled, the column with edit button is hidden
- Demo with feature added
- Check if editor is dirty if user tries to load another item or close the current one by performing a second click on the row/edit button.

Fixes #168 and #170

---
This change is [Reviewable](https://reviewable.io/reviews/vaadin/vaadin-crud/173)

### CODE REVIEW

**Comment by DiegoCardoso** [comment]
@Haprog there was no decision about the name and it's something that needs to be defined. I like your options. I will need to ask around about this matter.

File: src/vaadin-crud-grid.html [hunk 1]
```diff
@@ -43,12 +43,16 @@
             /**
              * Do not add headers to columns.
              */
-            noHead: Boolean
+            noHead: Boolean,
+            hideEditColumn: Boolean
```
Comment (yuriy-fix) [inline src/vaadin-crud-grid.html]: Not sure, but I guess this one can be private.

File: src/vaadin-crud-grid.html [hunk 2]
```diff
@@ -57,6 +61,21 @@
           }
         }
 
+        __onHideEditColumnChange(hideEditColumn) {
+          if (!this.innerHTML == '') {
```
Comment (yuriy-fix) [inline src/vaadin-crud-grid.html]: Nit: not sure if it's bulletproof. Can we use smth like `!this.firstChild`? 🤔

File: src/vaadin-crud-grid.html (hunk 1 above)
Comment (DiegoCardoso) [reply src/vaadin-crud-grid.html]: Now it's private.

File: src/vaadin-crud-grid.html (hunk 2 above)
Comment (DiegoCardoso) [reply src/vaadin-crud-grid.html]: I used the same check that is being used a bit down on the code, but changed to use the suggested one.

File: src/vaadin-crud.html [hunk 3]
```diff
@@ -415,6 +426,7 @@ <h3 slot="header">[[__editorHeader]]</h3>
           this.__editListener = e => this.__onCrudGridEdit(e);
           this.__changeListener = e => this.__onFormChanges(e);
           this.__gridSizeListener = this.__onGridSizeChanges.bind(this);
+          this.__clickRowToEditListener = this.__clickRowToEditListener.bind(this);
```
Comment (yuriy-fix) [inline src/vaadin-crud.html]: Not sure, but I am slightly afraid that there could be a case when it won't remove the listener (cause of the same name used). Haven't tested though.

We can use the same approach as in grid: https://github.com/vaadin/vaadin-grid/blob/51240f4fbcff7ad0f527add14e34ed3cb5f57875/src/vaadin-grid-selection-column.html#L148
just to be on safe side. (use `_bound` prefix)

File: src/vaadin-crud.html [hunk 4]
```diff
@@ -595,8 +655,11 @@ <h3 slot="header">[[__editorHeader]]</h3>
 
         __closeEditor() {
           this.editorOpened = false;
+          this.__isDirty = false;
           // Delay changing the item in order not to modify editor while closing
-          setTimeout(() => this.editedItem = undefined);
+          setTimeout(() => {
```
Comment (yuriy-fix) [inline src/vaadin-crud.html]: Nit: can still be one-liner 😅

File: src/vaadin-crud.html (hunk 3 above)
Comment (DiegoCardoso) [reply src/vaadin-crud.html]: Done.

File: src/vaadin-crud.html (hunk 4 above)
Comment (DiegoCardoso) [reply src/vaadin-crud.html]: Done.

**Pull request merged**